build-tools/build.sh

```diff
@@ -0,0 +1,230 @@
+    # Add to Workflow summary
+    echo "# ::error::You must provide a library folder name if copy_readme_and_license_to_include is true" >>$GITHUB_STEP_SUMMARY
+    exit 102502 # This is the string "pros-build" turned into int values, added together, and then multiplied by 10 plus the error code at the end. This is to hopefully avoid conflicts with other error codes.
+fi
+
+# Multithreading
+if [[ "$INPUT_MULTITHREADING" == "true" ]]; then
+    echo "Multithreading is enabled"
+    make_args="-j"
+else
+    echo "Multithreading is disabled"
+    make_args=""
+fi
+
+# ------------
+# ECHO LICENSE
+# ------------
+echo "::group::License"
+cat /LICENSE
+echo "::endgroup::"
+
+# ----------------
+# SETTING VARIABLES
+# ----------------
+
+set -e # Exit on error
+
+git config --global --add safe.directory /github/workspace
+
+# ----------------
+# CHECK IF TEMPLATE
+# ----------------
+echo "::group::Checking if this is a template"
+
+template=$(awk -F'=' '/^IS_LIBRARY:=/{print $2}' Makefile)
+if [ "$template" == "1" ]; then
+    echo "is template"
+else
+    echo "is not template"
+fi
+
+echo "::endgroup::"
+
+# ----------------
+# GET PROJECT INFO
+# ----------------
+echo "::group::Getting project info"
+
+if [ "$GITHUB_EVENT_NAME" == "pull_request" -o "$GH_EVENT_NAME" == "pull_request" ]; then
+    # Fetch the head SHA directly from the PR API
+    # if /$GITHUB_PR_NUM is "" check $INPUT_GITHUB_PR_NUM
+    if [ -z "$GITHUB_PR_NUM" ]; then
+        GITHUB_PR_NUM=$GH_PR_NUM
+    fi
+    API_URL="https://api.github.com/repos/$GITHUB_REPOSITORY/pulls/$GITHUB_PR_NUM"
+    echo "API URL: $API_URL"
+    API_RESPONSE=$(wget -O- --quiet "$API_URL")
+    if [ $? -ne 0 ]; then
+        echo "Error fetching data from GitHub API"
+        exit $?
+    fi
+    echo "API Response: $API_RESPONSE"
+    sha=$(echo "$API_RESPONSE" | jq -r '.head.sha' | head -c 6)
+
+else
+    # Use the commit SHA after the event
+    sha=$(git rev-parse HEAD | head -c 6)
+fi
+
+echo "SHA found: $sha"
+
+version=$(awk -F'=' '/^VERSION:=/{print $2}' Makefile)
+echo "Version found: $version"
+
+library_name=$(awk -F'=' '/^LIBNAME:=/{print $2}' Makefile)
+echo "Library name found: $library_name"
+
+echo "Version before setting postfix: $version"
+echo "SHA before setting postfix: $sha"
+
+if [ "$INPUT_NO_COMMIT_HASH" == "true" ]; then
+    postfix="${version}"
+else
+    postfix="${version}+${sha}"
+fi
+echo "Postfix after setting: $postfix"
+
+name="$library_name@$postfix"
+echo "name=$name" >>"$GITHUB_OUTPUT"
+echo "Name found: $name"
+
+echo "::endgroup::"
+# ----------------
+# BUILDING PROJECT
+# ----------------
+# Pause errors
+set +e
+make clean $make_args
+STD_OUTPUT=$(mktemp)
+# Set IS_LIBRARY to 0 to build the project if $template is 1
+if (($template == 1)); then
+    echo "::group::Building ${name}"
+    echo "Setting IS_LIBRARY to 0"
+    sed -i "s/^IS_LIBRARY:=.*\$/IS_LIBRARY:=0/" Makefile
+fi
```

Comment:
why?

Reply:
Why what?

Reply:
why modify IS_LIBRARY?

Reply:
So it would actually build from scratch. (That's the only thing that worked for me)

Reply:
ig if it works, then its fine